hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/webapp/TestRMWebServicesNodeLabels.java

```diff
@@ -119,527 +132,441 @@ public TestRMWebServicesNodeLabels() {
+    NodeLabelInfo nodeLabelInfo = new NodeLabelInfo(nlInfo.getLeft(), nlInfo.getRight());
+    assertTrue(String.format("Cannot find nodeLabelInfo '%s' among items of node label info list:" +
+            " %s", nodeLabelInfo, nodeLabelsInfo.getNodeLabelsInfo()),
+        nodeLabelsInfo.getNodeLabelsInfo().contains(nodeLabelInfo));
+  }
+
+  private void assertNodeLabelsInfoDoesNotContain(NodeLabelsInfo nodeLabelsInfo, Pair<String,
+      Boolean> nlInfo) {
+    NodeLabelInfo nodeLabelInfo = new NodeLabelInfo(nlInfo.getLeft(), nlInfo.getRight());
+    assertFalse(String.format("Should have not found nodeLabelInfo '%s' among " +
+        "items of node label info list: %s", nodeLabelInfo, nodeLabelsInfo.getNodeLabelsInfo()),
+        nodeLabelsInfo.getNodeLabelsInfo().contains(nodeLabelInfo));
+  }
+
+  private void assertNodeLabelsSize(NodeLabelsInfo nodeLabelsInfo, int expectedSize) {
+    assertEquals(expectedSize, nodeLabelsInfo.getNodeLabelsInfo().size());
+  }
+
+  private ClientResponse replaceNodeToLabels(WebResource r,
+      List<Pair<String, List<String>>> nodeToLabelInfos) throws Exception {
+    NodeToLabelsEntryList nodeToLabelsEntries = new NodeToLabelsEntryList();
+
+    for (Pair<String, List<String>> nodeToLabelInfo : nodeToLabelInfos) {
+      ArrayList<String> labelList = new ArrayList<>(nodeToLabelInfo.getRight());
+      String nodeId = nodeToLabelInfo.getLeft();
+      NodeToLabelsEntry nli = new NodeToLabelsEntry(nodeId, labelList);
+      nodeToLabelsEntries.getNodeToLabels().add(nli);
+    }
+    return r.path("ws").path("v1").path("cluster")
+        .path("replace-node-to-labels")
+        .queryParam("user.name", userName)
+        .accept(MediaType.APPLICATION_JSON)
+        .entity(toJson(nodeToLabelsEntries, NodeToLabelsEntryList.class),
+            MediaType.APPLICATION_JSON)
+        .post(ClientResponse.class);
+  }
+
+  private ClientResponse getNodeLabelMappings(WebResource r) {
+    ClientResponse response;
     response =
         r.path("ws").path("v1").path("cluster")
-            .path("get-node-labels").queryParam("user.name", userName)
+            .path("label-mappings").queryParam("user.name", userName)
             .accept(MediaType.APPLICATION_JSON).get(ClientResponse.class);
+    return response;
+  }
+
+  private ClientResponse getNodeLabelMappingsByLabels(WebResource r, String... labelNames) {
+    MultivaluedMapImpl params = createMultiValuedMap(labelNames);
+    return r.path("ws").path("v1").path("cluster")
+        .path("label-mappings").queryParam("user.name", userName)
+        .queryParams(params)
+        .accept(MediaType.APPLICATION_JSON).get(ClientResponse.class);
+  }
+
+  private ClientResponse replaceLabelsOnNode(WebResource r, String node, String... labelNames) {
+    return replaceLabelsOnNodeWithUserName(r, node, userName, labelNames);
+  }
+
+  private ClientResponse replaceLabelsOnNodeWithUserName(WebResource r, String node,
+      String userName, String... labelNames) {
```

Comment:
Check style is complaining that userName hides a field, just tweaking the name to user or similar would work.
The other question is if most of these could be static.

Reply:
As "replaceLabelsOnNode" calls "replaceLabelsOnNodeWithUserName" with various parameters, I don't think it's worth to declare all parameters as static.
For the checkstyle issue: I think it's not a useful check what checkstyle does in this case and I don't think I can come up with another name that is different but still descriptive. Are you okay with ignoring this checkstyle issue?
I'm usually ignoring this kind of "issue" from checkstyle when committing. 
Thanks.